templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -25,16 +25,44 @@
         {block name='product_cover'}
           {foreach from=$product.images item=image key=key}
             <div class="carousel-item{if $image.id_image == $product.default_image.id_image} active{/if}">
-              <img
-                class="img-fluid"
-                src="{$image.bySize.large_default.url}"
-                {if !empty($image.legend)}
-                  alt="{$image.legend}"
-                  title="{$image.legend}"
-                {else}
-                  alt="{$product.name}"
+              <picture>
+                {if isset($image.bySize.default_320.sources.avif)}
+                  <source 
+                    srcset="
+                      {$image.bySize.default_320.sources.avif} 320w,
+                      {$image.bySize.default_720.sources.avif} 720w,
+                      {$image.bySize.default_1440.sources.avif} 1440w"
+                    sizes="(min-width: 1300px) 720px, (min-width: 768px) 50vw, 100vw" 
+                    type="image/avif"
+                  >
                 {/if}
-                loading="lazy">
+
+                {if isset($image.bySize.default_320.sources.webp)}
+                  <source 
+                    srcset="
+                      {$image.bySize.default_320.sources.webp} 320w,
+                      {$image.bySize.default_720.sources.webp} 720w,
+                      {$image.bySize.default_1440.sources.webp} 1440w"
+                    sizes="(min-width: 1300px) 720px, (min-width: 768px) 50vw, 100vw" 
+                    type="image/webp"
+                  >
+                {/if}
+
+                <img
+                  class="img-fluid"
+                  srcset="
+                    {$image.bySize.default_320.url} 320w,
+                    {$image.bySize.default_720.url} 720w,
+                    {$image.bySize.default_1440.url} 1440w"
+                  sizes="(min-width: 1300px) 720px, (min-width: 768px) 50vw, 100vw" 
+                  src="{$image.bySize.default_720.sources.jpg}" 
+                  width="{$image.bySize.default_720.width}"
+                  height="{$image.bySize.default_720.height}"
+                  loading="lazy"
```

Comment:
We need to use `loading="eager"` on the first image. It will be FCP element that we need to load ASAP. It would maybe be also good to add a preload element to header so it starts loading right away.

Something like:
```
<link rel="preload" as="image"
  imagesrcset="{$product.cover.bySize.medium_default.url} 452w,
  {$product.cover.bySize.pdt_180.url} 180w,
  {$product.cover.bySize.pdt_300.url} 300w,
  {$product.cover.bySize.pdt_360.url} 360w,
  {$product.cover.bySize.pdt_540.url} 540w,
  {$product.cover.bySize.pdt_590.url} 590w"
  imagesizes="(min-width: 1300px) 590px, (min-width: 768px) 50vw, 100vw"
  href="{$product.cover.bySize.medium_default.url}">
```

Reply:
Leaving the comment opened for the preload if we want to implement it